### PR TITLE
Add examples as playbook for each module SWIM, Site and Inventory, Fi…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…x the issue of updating snmp_protocol from ssh  to telnet with netconf port, providing support to add multiple User Defined Field to single device.

In this commit -

— Add examples as playbook for each module SWIM, Site and Inventory in the module documentation in both intent and workflow manager modules.

— Fix the issue of updating snmp_protocol from ssh  to telnet with netconf port.

— Enhanced the functionality by providing support to add multiple User Defined Field to single device.

— Changed parameters from dict to list of string while exporting device details.

### CODE REVIEW

File: plugins/modules/inventory_intent.py [hunk 1]
```diff
@@ -1916,14 +1885,16 @@ def get_udf_id(self, field_name):
...
             response = self.dnac._exec(
                 family="devices",
                 function='get_all_user_defined_fields',
                 params={"name": field_name},
             )
             self.log("Received API response from 'get_all_user_defined_fields': {0}".format(str(response)), "DEBUG")
             udf = response.get("response")
-            udf_id = udf[0].get("id")
+            if udf:
+                udf_id = udf[0].get("id")
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:1898]: what is the value of udf_id if udf is not there.. We will return udf_id.. Can you please check it?

File: plugins/modules/inventory_intent.py [hunk 2]
```diff
@@ -2989,40 +2929,42 @@ def get_diff_merged(self, config):
...
-                # Create the Global UDF
-                self.create_user_defined_field().check_return_status()
+                # Check if the Global User defined field exist if not then create it with given field name
+                udf_exist = self.is_udf_exist(field_name)
 
-            # Get device Id based on config priority
-            device_ips = self.get_device_ips_from_config_priority()
-            device_ids = self.get_device_ids(device_ips)
+                if not udf_exist:
+                    # Create the Global UDF
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:2948]: can you log a debug message here?

File: plugins/modules/inventory_intent.py [hunk 3]
```diff
@@ -2807,6 +2744,9 @@ def get_diff_merged(self, config):
                     if playbook_params['netconfPort'] == " ":
                         playbook_params['netconfPort'] = None
 
+                    if playbook_params['netconfPort'] and playbook_params['cliTransport'] == "telnet":
+                        playbook_params['netconfPort'] = None
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:2748]: can you add a debug log message?

File: plugins/modules/inventory_intent.py [hunk 4]
```diff
@@ -2989,40 +2929,42 @@ def get_diff_merged(self, config):
...
-                self.result['response'] = self.msg
-                self.log(self.msg, "INFO")
-                return self
+                # Get device Id based on config priority
+                device_ips = self.get_device_ips_from_config_priority()
+                device_ids = self.get_device_ids(device_ips)
+
+                if len(device_ids) == 0:
+                    self.status = "failed"
+                    self.msg = "Can't Assign Global User Defined Field to device as device's are not present in Cisco Catalyst Center"
```
Comment (madhansansel) [inline plugins/modules/inventory_intent.py:2957]: self.msg = "Unable to assign Global User Defined Field: No devices found in Cisco Catalyst Center. Please add devices to proceed."

File: plugins/modules/inventory_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:1898]: we are returning udf_id as None then we check udf_id from the api caller place.

File: plugins/modules/inventory_intent.py (hunk 4 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:2957]: added.

File: plugins/modules/inventory_intent.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:2748]: added.

File: plugins/modules/inventory_intent.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/inventory_intent.py:2948]: added.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**